src/web/auth.rs

```diff
@@ -24,6 +24,42 @@ use time::Duration;
 use server::ServerState;
 use settings::{Secrets, ServerLimits, Settings};
 
+/// Represents a user-identifier that is extract from the authentication token
+///
+/// This token should be adapted as needed for the storage system to store data
+/// for the user.
+#[derive(Debug, Default, Deserialize, PartialEq, Serialize)]
+pub struct HawkIdentifier {
+    /// For MySQL database backends as the primary key
+    pub legacy_id: u64,
+    /// For NoSQL database backends that require randomly distributed primary keys
+    pub fxa_id: String,
```

Comment:
I'm a bit confused by the naming here, maybe we can chat about it in this morning's meeting?

But anyway, fwiw there is also an `fxa_uid` property available in the `HawkPayload`, which I didn't bother deserialising because I didn't have an immediate use for. Does this risk confusion with that, or could we even just use that instead?

Reply:
> I'm a bit confused...

Oh wait, ignore me, it's early here and I just woke up. This is all realted to mozilla-services/tokenserver#121 isn't it? I just remembered it's a thing.